### DISCUSSION TITLE
how to connect orange to SQL server

### DISCUSSION THREAD

**Discussion by kixxra** [biolab/orange3#6165] [Q&A]
hi, 

I've tried to connect orange with mySQL server. I've already installed pymsmsql 2.2.5 on my PC, and still can't connect to SQL Server.

btw, my phyton version is 3.7.2. Can it connect to orange? 
Or do I need to downgrade my phyton version??

thanks.

**Comment by ajdapretnar** [comment]
We are missing some details. How did you install Orange? Which version of Orange do you have?

**Reply by kixxra** [reply]
i'd downloaded it from the official site (https://orangedatamining.com/download/#windows), ver.3.33.

**Reply by ajdapretnar** [reply]
Right. So my guess would be you didn't install `pymssql` in the proper Orange environment. You need to open Orange Command Prompt and run `pip install pymssql` in there.

**Reply by kixxra** [reply]
oh i see, done it, but still got the same error message "please install a backend to use this widget"
![Capture image](https://user-images.githubusercontent.com/105841443/195034532-df93e688-8c5b-4fe3-8867-735d67b4ea79.JPG)

**Reply by liosc** [reply]
> Right. So my guess would be you didn't install `pymssql` in the proper Orange environment. You need to open Orange Command Prompt and run `pip install pymssql` in there.

Could you please explain how to "open Orange Command Prompt" to install pymssql in a way that is visible to Orange? I installed the package, it is in appdata\roaming\python\python313\site-packages folder but Orange keeps telling me to install a backend.

**Reply by ajdapretnar** [reply]
@liosc Orange Command Prompt is available only on Windows when installed with .exe. If you installed the package, then you can install `pymssql` directly into the virtual environment you installed Orange in.

**Reply by liosc** [reply]
Thank you @ajdapretnar 
I am on Windows but using the portable version. Can you tell me how to make available pymssql in this scenario?

**Reply by ajdapretnar** [reply]
@thocevar Would you happen to know?

**Reply by thocevar** [reply]
Use Command Line to move into the Orange folder of the extracted portable package (`...\Orange3-3.38.1\Orange`). This way running the command `python` will pick up the python located in that folder (you can check this by running `where python`, which should list this location at the top). You can use it to install `pymssql ` with `python -m pip install pymssql`. This installs the pymssql package into the python distribution that came with the portable package.

**Reply by liosc** [reply]
It worked. Thank you very much @thocevar @ajdapretnar
